2.crud_queries/tasks.md

```diff
@@ -61,11 +61,11 @@ db.articles.update({"title": "Node"}, {$push: {tag: 'nodemon'}});
 db.articles.update({"title" : "Node"}, {$set : {"tag": 'changed tag'}});
 
 
-db.articles.update({"title" : "Node"}, {"tag": 'changed tag'}});
+db.articles.update({"title" : "Node"}, {$push : {"tag": "HTML"}});
 ```
   - Write the differences here ?
 
-When we use {$set} it updates without changing the original data. When we update it without using {$set} we change the entire data for that field.
+When we use `{$set}` it updates the entire field. However when we use `{$push}` we update field without changing the pre-existing data.
 
```

Comment:
I am only saying to differentiate between using `$set` and without `$set`. `$push` is simply used to update field which has array of strings or number.

For example, we have a user
```js
{
  name: "",
  email: "",
  sports: ["js", "node"]
}

Now, if we have to update only `name` field, we should use `$set`.
Our query could be:-
```
db.users.update({}, {$set: {name: "some new name"}})
```

If I update it without `$set`, it will remove all the fields from the document and replace it with fields present in update clause.

Before update the document is:-
```js
{
  name: "",
  email: "",
  sports: ["js", "node"]
}
```
After running update operation without $set
```
db.users.update({}, {name: "some new name"})
```

Our document becomes
{
name: "some new name"
}